### PR TITLE
Speed up windows package lookup

### PR DESCRIPTION
**Pull request by ksubrama**

### CODE REVIEW

**Comment by ksubrama** [comment]
This is supposed to fix https://github.com/chef/inspec/issues/86 but I can't figure out what the unit tests are doing and what and _() function is....  I'll look into this tomorrow.

File: test/unit/resources/package_test.rb [hunk 1]
```diff
@@ -53,8 +53,11 @@
 
   # windows
   it 'verify windows package parsing' do
+    STDERR.puts "FUCK EVERYTHING!"
```
Comment (stevendanna) [inline test/unit/resources/package_test.rb]: We probably don't want to commit this :D

Comment (chris-rock) [reply test/unit/resources/package_test.rb]: @ksubrama Its not acceptable to place offensive language in a PR.

**Comment by chris-rock** [comment]
@ksubrama thanks for the new implementation. I added unit and integration tests for that implementation

File: test/integration/default/package_spec.rb [hunk 2]
```diff
@@ -28,3 +28,31 @@
...
+# the following test will iterate over all packages retrieved via WMI and verifies that the
+# optimized package implementation is returning the same results
+if os.windows?
+  # compare optimized version with wmi results
+  packages = powershell("Get-WmiObject -Class Win32_Product |  % { $_.Name }").stdout.strip.split("\n")
+
+  packages.each { |pkg|
+    package_name = pkg.strip
+    # get wmi package information
+    cmd = powershell("Get-WmiObject -Class Win32_Product | Where-Object {$_.Name -eq '#{package_name}'} | Select-Object -Property Name,Version,Vendor,PackageCode,Caption,Description | ConvertTo-Json")
```
Comment (stevendanna) [inline test/integration/default/package_spec.rb:41]: 👍 I like that we are using the old command here to ensure we match the same output we would have gotten with the old code.

**Comment by stevendanna** [comment]
👍 This looks reasonable to me.

**Comment by chris-rock** [comment]
fixes https://github.com/chef/inspec/issues/86

**Pull request merged**